code/test/util.js

```diff
@@ -257,7 +427,234 @@ const printInnerTrees = ({ Debugger, innerTrees }) => {
   }
 }
 
+// makeWallet uses an index and unlocked web3.eth.account and creates and funds a ONEwallet
+const makeWallet = async (salt, deployer, effectiveTime, duration, setLastResortAddress) => {
+  if (duration === undefined) { duration = DURATION }
+  let lastResortAccount = web3.eth.accounts.create()
+  let lastResortAddress = lastResortAccount.address
+  if (setLastResortAddress === false) { lastResortAddress = ONEConstants.EmptyAddress }
+  const { wallet, seed, hseed, root, client: { layers } } = await createWallet({
+    salt: new BN(ONEUtil.keccak(salt)),
+    effectiveTime,
+    duration,
+    maxOperationsPerInterval: SLOT_SIZE,
+    lastResortAddress: lastResortAddress,
+    spendingLimit: ONE_ETH
+  })
+  // Fund wallet
+  await web3.eth.sendTransaction({
+    from: deployer,
+    to: wallet.address,
+    value: HALF_ETH
+  })
+  return { wallet, seed, hseed, root, layers, lastResortAddress: lastResortAccount.address }
```

Comment:
just lastResortAddress

Reply:
seed, hseed, root, layers are already in wallet

Reply:
Thanks for the clarification 🙏 
I have removed `root`
This code was based on `basic.js` and the parameters are used for generating One Time Passwords and calling the transactions
They are used in `executeStandardTransaction` and will need to re replaced before removing.
seed : `const otp = ONEUtil.genOTP({ seed: walletInfo.seed, counter })`
hseed: `const eotp = await ONE.computeEOTP({ otp, hseed: walletInfo.hseed })`
layers : 
```
 let { tx, authParams, revealParams: returnedRevealParams } = await commitReveal({
    Debugger,
    layers: walletInfo.layers,
    index,
    eotp,
    paramsHash,
    commitParams,
    revealParams,
    wallet: walletInfo.wallet
  })
```



If I understand this correctly
ONEWallet.sol calls Executor.sol using
Executor.execute(op, tokenTrackerState, backlinkAddresses, signatures, spendingState);
but only receives in the OperationParameters in the reveal call
The rest come from the ONEWallets state
e.g.
IONEWallet[] backlinkAddresses; // to be set in next version - these are addresses forwarding funds and tokens to this contract AND must have their forwardAddress updated if this contract's forwardAddress is set or updated. One example of such an address is a previous version of the wallet "upgrading" to a new version. See more at https://github.com/polymorpher/one-wallet/issues/78
So we should only ever pass Operation Paramaeters
Even in commitParams and revealParams`